backend/apps/core/cache/invalidation.py

```
@@ -0,0 +1,119 @@
+
+logger = logging.getLogger(__name__)
+
+def add_key_to_tag(tag: str, key: str):
+    """
+    Registers a cache key under a tag in Redis (or LocMem fallback).
+    """
+    try:
+        from django_redis import get_redis_connection
+        con = get_redis_connection("default")
+        con.sadd(f"cache_tag:{tag}", key)
+    except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)